components/BaselayerSwitcher.svelte

```diff
@@ -0,0 +1,30 @@
+<script>
+  export let style;
+
+  function changeStyle() {
```

Comment:
When I've tried using `setStyle` before to do this without replacing the entire page, I've hit problems. It removes all sources and layers from the map, because maplibre doesn't distinguish the "basemap" from all the things we add on top. https://stackoverflow.com/a/36169495 has a workaround -- when `map.on('style.load'` happens, setup all the sources and layers. There's not an easy way for us to do that yet. So for now, get things working by just jumping to a new page, stashing the style in the URL. Clunky, but working